clang/include/clang/3C/ConstraintVariables.h

```diff
@@ -181,6 +181,13 @@ bool isAValidPVConstraint(const ConstraintVariable *C);
 class PointerVariableConstraint;
 class FunctionVariableConstraint;
 
+struct InternalTypedefInfo { 
+	bool hasTypedef;
+	int typedefLevel;
```

Comment:
What if you have a typedef of a typedef? 
```
typedef int *ptr_int_t;
typedef ptr_int_t Ptr_int_t;
void foo(void) {
  Ptr_int_t x = 0;
}
```
Maybe I'm not understanding what a "level" is, in this data structure?

Reply:
The level is where in the type tree the first typedef appears. This is because we only want to rewrite to the first typedef. Using your example when we rewrite the type `Ptr_int_t`, we want to stop immediately, as we've already hit the first typedef. (`ptr_int_t` still needs needs to be considered for constraint solving, which this does not affect)